Port moderation controllers to Pyramid

charmander opened the pull request:
Did I do it right?

charmander committed:
Make moderator_only and admin_only imply login_required

charmander committed:
Port moderation controllers to Pyramid

hyena commented on weasyl/controllers/decorators.py:
```diff
@@ -36,7 +36,7 @@ def inner(request):
             return Response(define.errorpage(request.userid, errorcode.permission))
         return view_callable(request)
-    return inner
+    return login_required(inner)
```
Both of these decorators now need to have docstrings that mention that they imply login required. It's too easy for someone glancing through and recognizing a decorator to gloss over the actual return line.

(also it'll show up in any docs we generate and in `help()` that way)

hyena commented on weasyl/controllers/moderation.py:
```diff
@@ -2,246 +2,205 @@
+
+@moderator_only
+@token_checked
+def modcontrol_suspenduser_(request):
```
nit: I've done a style of appending post_ to the post methods when there's also a get method. But since there's only two, it's no big deal. Should we change the other instances to this style of no `post_` on urls that are the 'default' behavior?

hyena commented on weasyl/controllers/moderation.py:
```diff
@@ -2,246 +2,205 @@
-        elif form.character:
-            moderation.unhidecharacter(int(form.character))
+    # TODO: Return a text response
+    return moderation.bulk_edit(
```
I think this will fail because it's not a Response object and the view callable isn't configured with a renderer. It should probably be wrapped in `Response()` here.

hyena commented on weasyl/controllers/routes.py:
```diff
@@ -207,6 +207,30 @@
+        'GET': moderation.modcontrol_suspenduser_get_,
+        'POST': moderation.modcontrol_suspenduser_,
+    }),
+    Route("/modcontrol/report", "modcontrol_report", {'GET': moderation.modcontrol_report_}),
```
If it's just a GET, it doesn't have to be a dict. Unless you have a stylistic objection to that shorthand.

hyena commented on weasyl/controllers/profile.py:
```diff
@@ -7,10 +7,7 @@
-    login_required,
-    moderator_only,
-)
+from weasyl.controllers.decorators import moderator_only
```
Love it.

hyena commented on weasyl/controllers/moderation.py:
```diff
@@ -2,246 +2,205 @@
+            moderation.removethumbnail(request.userid, submitid)
+        else:
+            raise WeasylError("Unexpected")
+        raise web.seeother("/submission/%i" % (submitid,))
```
Use the Pyramid Exception instead.

I also recommend searching for `web.` to find other places where this is used. Or better yet: Removing the import and checking for errors.

charmander committed:
Avoid mucking with the request in moderation.bulk_edit()

It’s quite inappropriate. (Well, most parts of this project are, but
this is taking it to new heights.)

charmander committed:
Remove leftover `web`

I searched the wrong file for it.

charmander committed:
Add get_ and post_ suffixes

charmander committed:
Remove unnecessary GET-only dict wrappers

hyena commented on weasyl/moderation.py:
```diff
@@ -608,8 +607,6 @@ def editcatchphrase(userid, otherid, content):
 
 
 def bulk_edit(userid, action, submissions=(), characters=(), journals=()):
-    web.header('Content-Type', 'text/plain')
```
The good news is that in my uncommitted index, there's a `request.set_header_on_response()` call. After I merge this, I can go back and make massaction use it.

charmander replied:
Why would you? It doesn’t seem right for this action to deal with response headers.
